src/utils.ts

```diff
@@ -113,6 +116,36 @@ export function getUnhardenedBIP32NodeToken(
   return `bip32:${index}`;
 }
 
+/**
+ * A hardened or unhardened BIP-32 node token, e.g. `bip32:0` or `bip32:0'`.
+ * Validates that the index is a non-negative integer number, and throws an
```

Comment:
I don't see where it's verified that the index is non-negative? 🤔

Reply:
Ah! Can we clarify that?

Reply:
Should we clarify implementation details in JSDoc comments?

Reply:
Hmm, actually a good question. I guess not? But it is a bit confusing to have the validation be performed by other functions called 🤔 

It might just be me 😅